data_model/src/lib.rs

```diff
@@ -1153,11 +1178,9 @@ where
     }
 }
 
-// TODO: This macro looks very similar to `from_and_try_from_value_identifiable`
-// and `from_and_try_from_value_identifiablebox` macros. It should be possible to
-// generalize them under one macro
+// TODO: The following macros looks very similar. Try to generalize them under one macro
```

Comment:
Honestly, don't think that that's a good idea. Generating stuff from macros itself impairs readability. Adding another layer makes it very difficult to reason about `macro_rules`. I would suggest instead turning these into `derive` macros, if possible.

Reply:
> Honestly, don't think that that's a good idea. Generating stuff from macros itself impairs readability

having worked with macros for a while, I generally agree with this sentiment. Not sure in this case